framework/src/task_manager/task_manager_core.c

```diff
@@ -754,18 +754,35 @@ static int taskmgr_getinfo_with_name(char *name, tm_response_t *response_msg)
 	response_msg->data = NULL;
 
 	for (chk_idx = 0; chk_idx < CONFIG_TASK_MANAGER_MAX_TASKS; chk_idx++) {
-		if (TM_LIST_ADDR(chk_idx) && ((strncmp(builtin_list[TM_IDX(chk_idx)].name, name, strlen(name)) == 0)
-					   || (strncmp(tm_task_list[TM_IDX(chk_idx)].name, name, strlen(name)) == 0)
+		if (TM_LIST_ADDR(chk_idx)) {
```

Comment:
Please refactor this logic like below:
```
char *thread_name = NULL;

if (builtin) {
   thread_name = builtin_list[ ].name;
} else if (task) {
   thread_name = tm_task_list[ ].name;
} else if (pthread) {
   thread_name = tm_pthread_list[ ].name;
}

if (thread_name && !strncmp(thread_name, name, strlen(name))) {
   ....
}
```

Reply:
Don't you need to use `strlen(name) + 1`?

Reply:
Done.